core/src/main/java/org/opensearch/sql/expression/function/PPLFuncImpTable.java

```diff
@@ -148,8 +171,61 @@ private abstract static class AbstractBuilder {
     abstract void register(BuiltinFunctionName functionName, FunctionImp functionImp);
 
     void registerOperator(BuiltinFunctionName functionName, SqlOperator operator) {
-      register(
-          functionName, (RexBuilder builder, RexNode... node) -> builder.makeCall(operator, node));
+      SqlOperandTypeChecker typeChecker;
+      if (operator instanceof SqlUserDefinedFunction udfOperator) {
+        typeChecker = extractTypeCheckerFromUDF(udfOperator);
+      } else {
+        typeChecker = operator.getOperandTypeChecker();
+      }
+
+      // Only the composite operand type checker for UDFs are concerned here.
+      if (operator instanceof SqlUserDefinedFunction
+          && typeChecker instanceof CompositeOperandTypeChecker compositeTypeChecker) {
+        register(functionName, createCompositeFunctionImp(operator, compositeTypeChecker));
+      } else if (typeChecker instanceof ImplicitCastOperandTypeChecker implicitCastTypeChecker) {
+        register(functionName, createImplicitCastFunctionImp(operator, implicitCastTypeChecker));
+      } else {
+        register(
+            functionName,
+            (RexBuilder builder, RexNode... node) -> builder.makeCall(operator, node));
+      }
+    }
+
+    private static SqlOperandTypeChecker extractTypeCheckerFromUDF(
+        SqlUserDefinedFunction udfOperator) {
+      UDFOperandMetadata udfOperandMetadata =
+          (UDFOperandMetadata) udfOperator.getOperandTypeChecker();
+      return (udfOperandMetadata == null) ? null : udfOperandMetadata.getInnerTypeChecker();
```

Comment:
SqlStdOperator does not have UDFOperandMetadata, so we will skip type checking? for instance SqlStdOperatorTable.AND, what is error message if the type is no correct?